### ISSUE TITLE
Does the Move operation returns correctly the new uids?

### ISSUE THREAD

**Issue by atiyka** [jstedfast/MailKit#1405]
**Describe the bug**
In case of Aol and Yahoo carriers, the `MoveTo` operation result is not correct. I'm not sure if the issue comes from them or from the Mailkit.
I move **two** messages from the inbox to a custom folder, let's say the uids in inbox are 186 and 187. The move returns the destination uids, which are (lets' say): 22 and 23. So, the 186 became 22 in the destination folder and the 187 became 23. But actually this is not corect, the order should be inverted.
When I debug the move operation, this is what I see: (check protocol logs as well below)
```
uniqueIdMap
	Destination	{22:23}
	Keys	{186:187}
	Source	{186:187}
	Values	{22:23}
		Count	2
		SortOrder	Ascending
Results View
	[0]	{[{186}, {22}]}
	[1]	{[{187}, {23}]}
```

I checked the same scenario with Gmail, and I found the following:
```
uniqueIdMap
	Destination	{9,8}
	Keys	{1747,1750}
	Source	{1747,1750}
	Values	{9,8}
		Count	2
		SortOrder	Descending
Results View
	[0]	{[{1747}, {9}]}
	[1]	{[{1750}, {8}]}
```
So basically the difference is that the sort order is Descending, and this way it sets correctly the new uids.

**Platform:**
 - OS: Windows
 - .NET Runtime: Core 3.1
 - MailKit Version: 2.11.1

**Protocol Logs**
```
C: X00000057 UID MOVE 187:186 xcc
S: * OK [COPYUID 1657263256 186:187 22:23]
S: * 159 EXPUNGE
S: * 159 EXPUNGE
S: X00000057 OK UID MOVE completed
```

Based on your experience, can I do simply to invert the values order in this case? Or what else could you suggest?

**Comment by jstedfast** [comment]
also, I would *always* use ascending ordered lists of indexes/uids when communicating with IMAP.

Most servers handle descending uids/indexes correctly, but some do not.

**Comment by jstedfast** [comment]
Based on the protocol log, the mapping is correct.

**Comment by atiyka** [comment]
> With the protocol logs you pasted, what makes you think they need to be inverted?

I think, because after the move operation, I fetch the messages from that folder, and in the `IMessageSummary` object the uids appears inverted, so for the message which was 186 and became 22 in the new folder based on the result of the move, currently it says that it has uid 23, and the other message has 22. So based on this I concluded that the move returned the result in a wrong order.

Anyway, thanks for your answer.

**Comment by jstedfast** [comment]
```
S: * OK [COPYUID 1657263256 186:187 22:23]
```

The above response is what gives MailKit's ImapClient implementation the UID mapping and as you can see, 186 maps to 22 and 187 maps to 23 according to the IMAP server.

**Comment by atiyka** [comment]
Okay, so I think the problem is on their end... Thank you
